docs/guide/bert_preprocessing_guide.ipynb

```diff
@@ -662,58 +661,61 @@
       "cell_type": "code",
       "execution_count": null,
       "source": [
-        "def bert_pretrain_preprocess(vocab_table, features):\n",
-        "  # Input is a string Tensor of documents, shape [batch, 1].\n",
-        "  text_a = features[\"text_a\"]\n",
-        "  text_b = features[\"text_b\"]\n",
-        "\n",
-        "  # Tokenize segments to shape [num_sentences, (num_words)] each.\n",
-        "  tokenizer = text.BertTokenizer(\n",
-        "      vocab_table,\n",
-        "      token_out_type=tf.int64)\n",
-        "  segments = [tokenizer.tokenize(text).merge_dims(\n",
-        "      1, -1) for text in (text_a, text_b)]\n",
-        "\n",
-        "  # Truncate inputs to a maximum length.\n",
-        "  trimmer = text.RoundRobinTrimmer(max_seq_length=6)\n",
-        "  trimmed_segments = trimmer.trim(segments)\n",
-        "\n",
-        "  # Combine segments, get segment ids and add special tokens.\n",
-        "  segments_combined, segment_ids = text.combine_segments(\n",
-        "      trimmed_segments,\n",
-        "      start_of_sequence_id=_START_TOKEN,\n",
-        "      end_of_segment_id=_END_TOKEN)\n",
-        "\n",
-        "  # Apply dynamic masking task.\n",
-        "  masked_input_ids, masked_lm_positions, masked_lm_ids = (\n",
-        "      text.mask_language_model(\n",
-        "        segments_combined,\n",
-        "        random_selector,\n",
-        "        mask_values_chooser,\n",
-        "      )\n",
-        "  )\n",
-        "  \n",
-        "  # Prepare and pad combined segment inputs\n",
-        "  input_word_ids, input_mask = text.pad_model_inputs(\n",
-        "    masked_input_ids, max_seq_length=_MAX_SEQ_LEN)\n",
-        "  input_type_ids, _ = text.pad_model_inputs(\n",
-        "    masked_input_ids, max_seq_length=_MAX_SEQ_LEN)\n",
-        "\n",
-        "  # Prepare and pad masking task inputs\n",
-        "  masked_lm_positions, masked_lm_weights = text.pad_model_inputs(\n",
-        "    masked_input_ids, max_seq_length=_MAX_PREDICTIONS_PER_BATCH)\n",
-        "  masked_lm_ids, _ = text.pad_model_inputs(\n",
-        "    masked_lm_ids, max_seq_length=_MAX_PREDICTIONS_PER_BATCH)\n",
-        "\n",
-        "  model_inputs = {\n",
-        "      \"input_word_ids\": input_word_ids,\n",
-        "      \"input_mask\": input_mask,\n",
-        "      \"input_type_ids\": input_type_ids,\n",
-        "      \"masked_lm_ids\": masked_lm_ids,\n",
-        "      \"masked_lm_positions\": masked_lm_positions,\n",
-        "      \"masked_lm_weights\": masked_lm_weights,\n",
-        "  }\n",
-        "  return model_inputs"
+        "def bert_pretrain_preprocess(vocab_table):\n",
+        "    def func(features):\n",
```

Comment:
nit: I find `bert_pretrain_preprocess` more readable as the preprocessing function (as opposed to a func that returns the preprocessing func). Can we reverse this and keep use of functools?